### PR TITLE
created subscription extensions

### PR DESCRIPTION
**Pull request by zikalino**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### CODE REVIEW

File: src/account/azext_account/azext_metadata.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "azext.isPreview": true,
```
Comment (yungezz) [inline src/account/azext_account/azext_metadata.json:2]: isExperimental

File: src/account/azext_account/generated/_help.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+from knack.help_files import helps
+
+
+helps['account subscription'] = """
+    type: group
+    short-summary: account subscription
+"""
+
+helps['account subscription create-subscription'] = """
```
Comment (yungezz) [inline src/account/azext_account/generated/_help.py:16]: create?

File: src/account/azext_account/generated/_help.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    type: command
+    short-summary: The operation to create a new WebDirect or EA Azure subscription.
+    examples:
+      - name: createSubscription
+        text: |-
+               az account subscription create-subscription --billing-account-name \\
```
Comment (yungezz) [inline src/account/azext_account/generated/_help.py:22]: could you pls use some meaningful value for example?

File: src/account/azext_account/generated/_help.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+               "0aa27f2b-ec7f-5a65-71f6-a5ff0897bd55:ae0dae1e-de9a-41f6-8257-76b055d98372_2019-05-31" \\
+               --billing-profile-name "27VR-HDWX-BG7-TGB" --cost-center "135366376" --display-name \\
+               "Contoso MCA subscription" --owner object-id=973034ff-acb7-409c-b731-e789672c7b32 \\
+               --sku-id "0001" --invoice-section-name "JGF7-NSBG-PJA-TGB"
+"""
+
+helps['account subscription create-subscription-in-enrollment-account'] = """
+    type: command
+    short-summary: Creates an Azure subscription
```
Comment (yungezz) [inline src/account/azext_account/generated/_help.py:31]: what's the differnece from above in short-description. seems the extension is purely generated

File: src/account/azext_account/generated/_help.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+      - name: createSubscription
+        text: |-
+               az account subscription create-subscription --billing-account-name \\
+               "0aa27f2b-ec7f-5a65-71f6-a5ff0897bd55:ae0dae1e-de9a-41f6-8257-76b055d98372_2019-05-31" \\
+               --billing-profile-name "27VR-HDWX-BG7-TGB" --cost-center "135366376" --display-name \\
+               "Contoso MCA subscription" --owner object-id=973034ff-acb7-409c-b731-e789672c7b32 \\
+               --sku-id "0001" --invoice-section-name "JGF7-NSBG-PJA-TGB"
+"""
+
+helps['account subscription create-subscription-in-enrollment-account'] = """
```
Comment (yungezz) [inline src/account/azext_account/generated/_help.py:29]: seperate a account type param?

File: src/account/azext_account/generated/_help.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+helps['account subscription rename'] = """
+    type: command
+    short-summary: The operation to rename a subscription
+    examples:
+      - name: renameSubscription
+        text: |-
+               az account subscription rename --subscription-id "83aa47df-e3e9-49ff-877b-94304bf3d3ad"
+"""
+
+helps['account subscription cancel'] = """
```
Comment (yungezz) [inline src/account/azext_account/generated/_help.py:63]: with confirmation?

File: src/account/azext_account/generated/_help.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+               az account subscription cancel --subscription-id "83aa47df-e3e9-49ff-877b-94304bf3d3ad"
+"""
+
+helps['account subscription enable'] = """
+    type: command
+    short-summary: Enable subscription
+    examples:
+      - name: Enable subscription
+        text: |-
+               az account subscription enable --subscription-id "7948bcee-488c-47ce-941c-38e20ede803d"
```
Comment (yungezz) [inline src/account/azext_account/generated/_help.py:79]: pls use fake subs id

File: src/account/azext_account/azext_metadata.json [hunk 8]
```diff
@@ -0,0 +1,4 @@
+{
+    "azext.isExperimental": true,
+    "azext.minCliCoreVersion": "2.0.67"
```
Comment (fengzhou-msft) [inline src/account/azext_account/azext_metadata.json:3]: minCliCoreVersion should also be modified to 2.3.1 to support Experimental tag.

File: src/account/setup.cfg [hunk 9]
```diff
@@ -0,0 +1,2 @@
+[bdist_wheel]
+universal=1
```
Comment (fengzhou-msft) [inline src/account/setup.cfg:2]: cloud be removed.

File: src/account/setup.py [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+# The full list of classifiers is available at
+# https://pypi.python.org/pypi?%3Aaction=list_classifiers
+CLASSIFIERS = [
+    'Development Status :: 4 - Beta',
+    'Intended Audience :: Developers',
+    'Intended Audience :: System Administrators',
+    'Programming Language :: Python',
+    'Programming Language :: Python :: 3',
+    'Programming Language :: Python :: 3.4',
+    'Programming Language :: Python :: 3.5',
```
Comment (fengzhou-msft) [inline src/account/setup.py:30]: 3.4, 3.5 could be removed.

File: src/account/azext_account/generated/_validators.py [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+    from msrestazure.tools import is_valid_resource_id, resource_id
+    if namespace.storage_account:
+        if not is_valid_resource_id(namespace.RESOURCE):
+            namespace.storage_account = resource_id(
+                subscription=get_subscription_id(cmd.cli_ctx),
+                resource_group=namespace.resource_group_name,
+                namespace='Microsoft.Storage',
+                type='storageAccounts',
+                name=namespace.storage_account
+            )
```
Comment (fengzhou-msft) [inline src/account/azext_account/generated/_validators.py:18]: Remove if not used.

File: src/account/azext_account/generated/custom.py [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+def account_subscription_create_subscription(cmd, client,
+                                             billing_account_name,
+                                             billing_profile_name,
+                                             invoice_section_name,
+                                             display_name,
+                                             sku_id,
+                                             cost_center=None,
+                                             owner=None,
+                                             management_group_id=None):
```
Comment (fengzhou-msft) [inline src/account/azext_account/generated/custom.py:20]: add `no_wait=False`

File: src/account/azext_account/generated/custom.py [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+
+    body = {}
+    body['display_name'] = display_name
+    body['sku_id'] = sku_id
+    body['cost_center'] = cost_center
+    body['owner'] = {'object_id': owner}
+    body['management_group_id'] = management_group_id
+    return client.create_subscription(billing_account_name=billing_account_name, billing_profile_name=billing_profile_name, invoice_section_name=invoice_section_name, body=body)
```
Comment (fengzhou-msft) [inline src/account/azext_account/generated/custom.py:28]: use sdk_no_wait.

File: src/account/azext_account/generated/action.py [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+        try:
+            properties = dict(x.split('=', 1) for x in values)
+        except ValueError:
+            raise CLIError('usage error: {} [KEY=VALUE ...]'.format(option_string))
+        d = {}
+        for k in properties:
+            kl = k.lower()
+            v = properties[k]
+            if kl == 'object_id':
+                d['object_id'] = v
```
Comment (fengzhou-msft) [inline src/account/azext_account/generated/action.py:28]: If the action only has one property `object_id`, we may pass it directly and can get rid of the action.

File: src/account/setup.cfg (hunk 9 above)
Comment (zikalino) [reply src/account/setup.cfg:2]: entire file?

Comment (fengzhou-msft) [reply src/account/setup.cfg:2]: You can keep the file for convenience of future editing.

**Comment by yonzhan** [comment]
subscription

File: src/account/azext_account/generated/action.py (hunk 14 above)
Comment (zikalino) [reply src/account/azext_account/generated/action.py:28]: yeah, that should be deleted, not used

**Comment by zikalino** [comment]
not sure why the test fail in firewall extension now

**Pull request merged**